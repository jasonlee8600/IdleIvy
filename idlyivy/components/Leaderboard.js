import Link from "next/link";

import React from 'react'
import {useEffect, useState, useRef} from 'react';
import LBHeader from "./LBHeader";
import LBList from "./LBList";



export default function Leaderboard() {
    const [list, setList] = useState([]);
    

    useEffect(()=>{
        fetch('http://localhost:3001/getLB')
          .then(response => response.json())
          .then(setList);
       }, []);

        // console.log("list is " + JSON.stringify(list))

        var top10list = JSON.parse(JSON.stringify(list))
        console.log(top10list)

       return (
                
                
            <div className="flex flex-col w-4/5 h-fit bg-[#DADADA] self-center items-center">  
                <LBHeader></LBHeader>
                
                <LBList leaders={top10list}></LBList>
            </div>
       
       );

}
