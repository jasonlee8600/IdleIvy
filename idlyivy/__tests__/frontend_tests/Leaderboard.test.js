import React from 'react';
import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom'
import Leaderboard from '../../components/Leaderboard';


afterEach(() => {
    cleanup(); 
})

describe("Leaderboard testing", () => {
    test('renders correctly', () => {
        const tree = renderer.create(<Leaderboard />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});