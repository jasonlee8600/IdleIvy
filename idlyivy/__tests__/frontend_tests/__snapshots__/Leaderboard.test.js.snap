// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard testing renders correctly 1`] = `
<div
  className="flex flex-col w-4/5 h-fit bg-gradient-to-b from-[#272727] to-[#797979] self-center items-center border-[5px] border-white"
  data-testid="leaderboard"
>
  <div
    className="font-display container h-14 content-center text-center grid grid-cols-11 bg-[#202B64] border-b-[3px] border-white"
  >
    <div
      className="tile col-span-5"
    >
      Nickname
    </div>
    <div
      className="tile col-span-5"
    >
      Rate
    </div>
  </div>
</div>
`;
